### PR TITLE
[NAE-1381] Public view task loading

### PR DESCRIPTION
**Pull request by Daniell7**
# Description

Set LoadingEmitter while system creating new case

Fixed [NAE-1381]

## Dependencies

none

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

Assisted testing + manual testing + front-end tests

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  | Windows 10 |
| Runtime             | Node 12.18.3 |
| Dependency Manager  | NPM 6.14.6 |
| Framework version   | Angular 10 |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @camperko
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1381]: https://netgrif.atlassian.net/browse/NAE-1381?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: projects/nae-example-app/src/app/doc/public-task-view/public-task-view.component.ts [hunk 1]
```diff
@@ -112,8 +116,13 @@ export class PublicTaskViewComponent extends AbstractTaskView implements AfterVi
 
     @ViewChild('header') public taskHeaderComponent: HeaderComponent;
 
-    constructor(taskViewService: TaskViewService) {
+    constructor(taskViewService: TaskViewService, publicTaskLoadingService: PublicTaskLoadingService) {
         super(taskViewService);
+        this.loading$ = combineLatest(taskViewService.loading$, publicTaskLoadingService.loading$).pipe(
+            map(sources => {
+                return !(!sources[0] && !sources[1]);
```
Comment (minop) [inline projects/nae-example-app/src/app/doc/public-task-view/public-task-view.component.ts:123]: would be more readable as `sources[0] || sources[1]`, but that doesn't seem to be equivalent. Is the expression correct?

Comment (Daniell7) [reply projects/nae-example-app/src/app/doc/public-task-view/public-task-view.component.ts:123]: Yes, expression is correct. I update it.

File: projects/nae-example-app/src/app/doc/public-task-view/public-task-view.component.ts [hunk 2]
```diff
@@ -112,8 +116,13 @@ export class PublicTaskViewComponent extends AbstractTaskView implements AfterVi
 
     @ViewChild('header') public taskHeaderComponent: HeaderComponent;
 
-    constructor(taskViewService: TaskViewService) {
+    constructor(taskViewService: TaskViewService, publicTaskLoadingService: PublicTaskLoadingService) {
```
Comment (machacjozef) [inline projects/nae-example-app/src/app/doc/public-task-view/public-task-view.component.ts:119]: Component?

**Review by machacjozef — CHANGES_REQUESTED** [review]

File: projects/nae-example-app/src/app/doc/public-task-view/public-task-view.component.ts (hunk 2 above)
Comment (Daniell7) [reply projects/nae-example-app/src/app/doc/public-task-view/public-task-view.component.ts:119]: The public task view does not have a separate component in the engine. So this implementation is within the constructor only in the example app. This part must be copied and pasted into the public task view in the project for proper functionality.

File: projects/netgrif-components-core/src/lib/view/task-view/service/public-task-loading.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  it('should set value', () => {
+        service.setLoading$(true);
+        expect(service.getEmitter().isActive).toBeTrue();
+  });
+
+  it('should get value', () => {
+        const loading = service.getEmitter().asObservable();
+        expect(service.loading$).toEqual(loading);
+    });
```
Comment (Kovy95) [inline projects/netgrif-components-core/src/lib/view/task-view/service/public-task-loading.service.spec.ts:29]: please add afterEach method to reset testing module, and fix intend

File: projects/netgrif-components-core/src/lib/view/task-view/service/public-task-loading.service.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import {LoadingEmitter} from '../../../utility/loading-emitter';
+import {Observable} from 'rxjs';
+
+@Injectable()
+export class PublicTaskLoadingService {
+
+    protected _loading$: LoadingEmitter;
+
+    constructor() {
+      this._loading$ = new LoadingEmitter();
```
Comment (Kovy95) [inline projects/netgrif-components-core/src/lib/view/task-view/service/public-task-loading.service.ts:11]: fix intend

File: projects/netgrif-components-core/src/lib/view/task-view/service/public-task-loading.service.spec.ts (hunk 3 above)
Comment (Daniell7) [reply projects/netgrif-components-core/src/lib/view/task-view/service/public-task-loading.service.spec.ts:29]: done

File: projects/netgrif-components-core/src/lib/view/task-view/service/public-task-loading.service.ts (hunk 4 above)
Comment (Daniell7) [reply projects/netgrif-components-core/src/lib/view/task-view/service/public-task-loading.service.ts]: done

**Review by minop — APPROVED** [review]

**Review by Kovy95 — APPROVED** [review]

**Pull request merged**